core/server/middleware/check-ssl.js

```diff
@@ -1,60 +1,69 @@
-var config = require('../config'),
-    url    = require('url'),
-    utils  = require('../utils'),
-    checkSSL;
-
-function isSSLrequired(isAdmin, configUrl, forceAdminSSL) {
-    var forceSSL = url.parse(configUrl).protocol === 'https:' ? true : false;
-    if (forceSSL || (isAdmin && forceAdminSSL)) {
-        return true;
-    }
-    return false;
-}
+var url = require('url'),
+    debug = require('debug')('ghost:redirects'),
+    utils = require('../utils'),
+    checkSSLAndRedirects;
+
+function redirectUrl(options) {
+    var redirectTo = options.redirectTo,
+        path = options.path,
+        query = options.query,
+        parts = url.parse(redirectTo);
 
-// The guts of checkSSL. Indicate forbidden or redirect according to configuration.
-// Required args: forceAdminSSL and url should be passed from config. reqURL from req.url
-function sslForbiddenOrRedirect(opt) {
-    var forceAdminSSL = opt.forceAdminSSL,
-        reqUrl        = url.parse(opt.reqUrl), // expected to be relative-to-root
-        baseUrl       = url.parse(opt.configUrl),
-        response = {
-        // Check if forceAdminSSL: { redirect: false } is set, which means
-        // we should just deny non-SSL access rather than redirect
-        isForbidden: (forceAdminSSL && forceAdminSSL.redirect !== undefined && !forceAdminSSL.redirect),
-
-        redirectUrl: function redirectUrl(query) {
-            return url.format({
-                protocol: 'https:',
-                hostname: baseUrl.hostname,
-                port: baseUrl.port,
-                pathname: reqUrl.pathname,
-                query: query
-            });
-        }
-    };
-
-    return response;
+    return url.format({
+        protocol: parts.protocol,
+        hostname: parts.hostname,
+        port: parts.port,
+        pathname: path,
+        query: query
+    });
 }
 
-// Check to see if we should use SSL
-// and redirect if needed
-checkSSL = function checkSSL(req, res, next) {
-    if (isSSLrequired(res.isAdmin, utils.url.urlFor('home', true), config.get('forceAdminSSL'))) {
-        if (!req.secure) {
-            var response = sslForbiddenOrRedirect({
-                forceAdminSSL: config.get('forceAdminSSL'),
-                configUrl: utils.url.urlFor('home', true),
-                reqUrl: req.originalUrl || req.url
-            });
-
-            if (response.isForbidden) {
-                return res.sendStatus(403);
-            } else {
-                return res.redirect(301, response.redirectUrl(req.query));
-            }
-        }
+/**
+ * SSL AND REDIRECTS
+ * @TODO: rename file
+ */
+checkSSLAndRedirects = function checkSSLAndRedirects(req, res, next) {
```

Comment:
I had to rewrite the check-ssl middleware (which is now called url-redirects middleware, see https://github.com/TryGhost/Ghost/pull/7937/commits/1894b6930c4c0be96beffe4a9115a655cba1c485)

### reasons for whole rewrite
The check ssl middleware did the following:

1. it checked if SSL is required based on `forceAdminSSL` and protocol of the defined url
2. it either has forbidden the request, because `forceAdminSSL.redirect` was defined or it redirected, because (1) was true

As there is no `forceAdminSSL` and `urlSSL` config option anymore, the whole logic did not fit into the new concept anymore.

Let me explain the concept.

You can define two urls

1. url of your blog
2. url of your admin/api

#### 2 rules
- redirect if url or admin host does not fit the requested host e.g. you request `localhost:2368/ghost`, but admin url is defined as `admin.localhost:2368/ghost`
- redirect if protocol does not match e.g. you request `http://localhost:2368`, but your blog ur is `https://localhost:2368/ghost`